From ff1567c2bfb6ab05289d4a1b2b1b6651b5902144 Mon Sep 17 00:00:00 2001
From: Sandor Yu <R01008@freescale.com>
Date: Wed, 11 Mar 2015 14:25:40 +0800
Subject: [PATCH] MLK-10402: v4l2 capture: Check NULL pointer before memory
 access

Function vb2_plane_vaddr() will return NULL if there is no
kernel virtual address mapping exist for a given plane.
Check the function return value in v4l2 capture driver
before access memory.

Signed-off-by: Sandor Yu <R01008@freescale.com>
---
 drivers/media/platform/mxc/subdev/mx6s_capture.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/mxc/subdev/mx6s_capture.c b/drivers/media/platform/mxc/subdev/mx6s_capture.c
index 8d5b06a..4e7cd0a 100644
--- a/drivers/media/platform/mxc/subdev/mx6s_capture.c
+++ b/drivers/media/platform/mxc/subdev/mx6s_capture.c
@@ -632,8 +632,9 @@ static int mx6s_videobuf_prepare(struct vb2_buffer *vb)
 	 * This can be useful if you want to see if we actually fill
 	 * the buffer with something
 	 */
-	memset((void *)vb2_plane_vaddr(vb, 0),
-	       0xaa, vb2_get_plane_payload(vb, 0));
+	if (vb2_plane_vaddr(vb, 0))
+		memset((void *)vb2_plane_vaddr(vb, 0),
+		       0xaa, vb2_get_plane_payload(vb, 0));
 #endif
 
 	vb2_set_plane_payload(vb, 0, csi_dev->pix.sizeimage);
-- 
1.8.0

